cumulusci/robotframework/CumulusCI.py

```diff
@@ -0,0 +1,160 @@
+        if org is None:
+            org = self.org
+        else:
+            org = self.config.keychain.get_org(org)
+        return org.start_url
+
+    def run_task(self, task_name, **options):
+        """ Runs a named CumulusCI task for the current project with optional
+            support for overriding task options via kwargs.
+            
+            Examples:
+            | =Keyword= | =task_name= | =task_options=             | =comment=                        |
+            | Run Task  | deploy      |                            | Run deploy with standard options |
+            | Run Task  | deploy      | path=path/to/some/metadata | Run deploy with custom path      |
+        """
+        task_config = getattr(self.config.project_config, 'tasks__{}'.format(task_name))
+        if not task_config:
+            raise TaskNotFoundError('Task not found: {}'.format(task_name))
+        class_path = task_config.get('class_path')
+        task_class, task_config = self._init_task(class_path, options, task_config)
+        return self._run_task(task_class, task_config)
+
+    def run_task_class(self, class_path, **options):
+        """ Runs a CumulusCI task class with task options via kwargs.
+
+            Use this keyword to run logic from CumulusCI tasks which have not
+            been configured in the project's cumulusci.yml file.  This is
+            most useful in cases where a test needs to use task logic for
+            logic unique to the test and thus not worth making into a named
+            task for the project
+
+            Examples:
+            | =Keyword=      | =task_class=                     | =task_options=                            |
+            | Run Task Class | cumulusci.task.utils.DownloadZip | url=http://test.com/test.zip dir=test_zip |
+        """ 
+        task_class, task_config = self._init_task(class_path, options, {})
+        return self._run_task(task_class, task_config)
+
+    def _init_api(self, base_url=None):
+        api_version = self.config.project_config.project__package__api_version
+
+        rv = Salesforce(
+            instance=self.org.instance_url.replace('https://', ''),
+            session_id=self.org.access_token,
+            version=api_version,
+        )
+        if base_url is not None:
+            rv.base_url += base_url
+        return rv
+
+    def _init_config(self):
+        config = CliConfig()
+        return config
+
+    def _init_task(self, class_path, options, task_config):
+        task_class = import_class(class_path)
+        task_config = self._parse_task_options(options, task_class, task_config)
+        return task_class, task_config
+
+    def _parse_task_options(self, options, task_class, task_config):
+        # Parse options and add to task config
+        if options:
+            if 'options' not in task_config:
+                task_config['options'] = {}
+            for name, value in options.items():
+                # Validate the option
```

Comment:
little surprised to see that BaseTask doesn't do this validation already. consider moving it there?